worktree.go

```diff
@@ -167,6 +170,61 @@ func (w *Worktree) getMode(fi billy.FileInfo) os.FileMode {
 	return object.FileMode
 }
 
+const gitmodulesFile = ".gitmodules"
+
+func (w *Worktree) Submodules() (Submodules, error) {
```

Comment:
document it, especially why it can return (empty slice and nil) and what does it means. Is it an error if the .gitmodules is empty?